src/Parser/VectorTileParser.js

```diff
@@ -130,41 +131,33 @@ function readPBF(file, options) {
     // Only if the layer.origin is top
     const y = options.in.isInverted ? options.extent.row : (1 << z) - options.extent.row - 1;
 
-    const collection = new FeatureCollection(options.out);
-
-    const vFeature = vectorTile.layers[sourceLayers[0]];
-    // TODO: verify if size is correct because is computed with only one feature (vFeature).
-    const size = vFeature.extent * 2 ** z;
+    const vFeature0 = vectorTile.layers[vtLayerNames[0]];
+    // TODO: verify if size is correct because is computed with only one feature (vFeature0).
+    const size = vFeature0.extent * 2 ** z;
     const center = -0.5 * size;
 
     collection.scale.set(globalExtent.x / size, -globalExtent.y / size, 1);
-    collection.position.set(vFeature.extent * x + center, vFeature.extent * y + center, 0).multiply(collection.scale);
+    collection.position.set(vFeature0.extent * x + center, vFeature0.extent * y + center, 0).multiply(collection.scale);
     collection.updateMatrixWorld();
 
-    sourceLayers.forEach((layer_id) => {
-        if (!options.in.layers[layer_id]) { return; }
+    vtLayerNames.forEach((vtLayerName) => {
+        if (!options.in.layers[vtLayerName]) { return Promise.resolve(collection); }
 
-        const sourceLayer = vectorTile.layers[layer_id];
+        const vectorTileLayer = vectorTile.layers[vtLayerName];
 
-        for (let i = sourceLayer.length - 1; i >= 0; i--) {
-            const vtFeature = sourceLayer.feature(i);
+        for (let i = vectorTileLayer.length - 1; i >= 0; i--) {
+            const vtFeature = vectorTileLayer.feature(i);
             vtFeature.tileNumbers = { x, y: options.extent.row, z };
-            const layers = options.in.layers[layer_id].filter(l => l.filterExpression.filter({ zoom: z }, vtFeature) && z >= l.zoom.min && z < l.zoom.max);
-            let feature;
+            // Find layers where this vtFeature is used
+            const layers = options.in.layers[vtLayerName]
+                .filter(l => l.filterExpression.filter({ zoom: z }, vtFeature));
 
             for (const layer of layers) {
-                if (!feature) {
-                    feature = collection.requestFeatureById(layer.id, vtFeature.type - 1);
-                    feature.id = layer.id;
-                    feature.order = layer.order;
-                    feature.style = options.in.styles[feature.id];
-                    vtFeatureToFeatureGeometry(vtFeature, feature);
-                } else if (!collection.features.find(f => f.id === layer.id)) {
-                    feature = collection.newFeatureByReference(feature);
-                    feature.id = layer.id;
-                    feature.order = layer.order;
```

Comment:
This seems like some kind of optimization, can you explain in more details why you remove it please?

Reply:
It's a bit weird that you get this warning since we look for features with the same id that would already have been parsed.
If we remove this code I'm not sure we need to features collection at all since from my understanding it is to avoid creating multiple times the same feature in memory. It would be interresting to check if it is common that same features are in different pbf and therefore what if this optimization is really needed. It seems a bit unrelated to this PR though so what I propose is to:
* keep this code but comment it for now
* add a comment above explaining what we've said here: this optimization is not fully working and needs to be re-assessed. It has been deactivated for now, see: https://github.com/iTowns/itowns/pull/2469/files#r1861802136